### PR TITLE
Moved `_state` from `_Synchronizer ` to `_ThreadPromise/_ThreadFuture`

### PR DESCRIPTION
**Pull request by christopherlagerhult**
Moved to develop branch (old: #1877)

### CODE REVIEW

**Comment by emilwestergren** [comment]
Reason for this?

**Comment by christopherlagerhult** [comment]
Same as #1877. `_state` is only used in `_ThreadPromise/_ThreadFuture` which might make `_state` ambiguous when dealing with Promises created elsewhere(ex. `_TaskPromise`). 

for example:
```ooc
_Synchronizer: abstract class {
	_state: _PromiseState
	...
	wait: abstract func (time: TimeSpan) -> Bool { 
            this wait(time)
            this _state == _PromiseState Finished 
}
```
A `_TaskPromise` will always return false here.
Might be stupid to change it but it is still weird to have a state that is always `Unfinished`.

File: source/concurrent/Promise.ooc [hunk 1]
```diff
@@ -118,6 +118,7 @@ Future: abstract class <T> extends _Synchronizer {
 }
 
 _ThreadFuture: class <T> extends Future<T> {
+	_state: _PromiseState
```
Comment (emilwestergren) [inline source/concurrent/Promise.ooc]: This should probably not be named `_PromiseState` if it's used for both `Future` and `Promise`. Not related to this PR though and not that important.

Comment (marcusnaslund) [inline source/concurrent/Promise.ooc]: #1889

**Comment by marcusnaslund** [comment]
I'm not completely sure about this, but it removes confusion as you say, and we can revisit it later if need be.

**Pull request merged**